lib/cadet_web/views/devices_view.ex

```diff
@@ -0,0 +1,16 @@
+defmodule CadetWeb.DevicesView do
+  use CadetWeb, :view
+
+  def render("index.json", %{registrations: registrations}) do
+    render_many(registrations, CadetWeb.DevicesView, "show.json", as: :registration)
+  end
+
+  def render("show.json", %{registration: registration}) do
+    %{
+      id: registration.id,
+      title: registration.title,
+      type: registration.device.type,
+      secret: registration.device.secret
+    }
```

Comment:
Whilst most of the other views use `transform_map_for_view` instead, this actually ends up much cleaner overall, so its fine.